### ISSUE TITLE
[button-has-type] Allow implicit 'submit' type

### ISSUE THREAD

**Issue by ocket8888** [angular-eslint/angular-eslint#1359]
## Description and reproduction of the issue

The rule appears to work just fine for me, but the suggestion I have is that an option be added that allows the use of implicit button typing. The default `type` of a `button` is `submit` if not specified (although possibly if the button isn't a descendant of a `form`? I'm unclear on that) and this rule will demand that I specify it explicitly. I'd prefer to only specify it as necessary; when it should be `button[type=button]`.

```JSON
{
  "rules": {
    "@angular-eslint/template/button-has-type": "error"
  }
}
```

```HTML
<form>
<!-- form content -->
<button>Submit</button>
</form>
```

## Versions

| package                                   | version |
| ----------------------------------        | ------- |
| `@angular-eslint/eslint-plugin-template`  | `14.1.2` |
| `@angular-eslint/template-parser`         | `14.1.2` |
| `@typescript-eslint/parser`               | `5.36.2` |
| `ESLint`                                  | `8.39.0` |
| `node`                                    | `18.15.0` |

```shellsession
$ ng version     

     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/
    

Angular CLI: 14.2.9
Node: 18.15.0 (Unsupported)
Package Manager: npm 9.5.0 
OS: linux x64

Angular: 14.2.11
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, platform-server
... router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1402.9
@angular-devkit/build-angular   14.2.9
@angular-devkit/core            14.2.9
@angular-devkit/schematics      14.2.9
@angular/cdk                    14.2.7
@angular/cli                    14.2.9
@angular/material               14.2.7
@nguniversal/express-engine     14.2.2
@schematics/angular             14.2.9
rxjs                            6.6.7
typescript                      4.8.4
    
Warning: The current version of Node (18.15.0) is not supported by Angular.
```

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

**Comment by JamesHenry** [comment]
Hi @ocket8888, this goes against the entire purpose of the rule - to always require an explicit type. The rule does not exist because button's always require explicit types, it exists to allow teams to enforce consistency and maximum clarity. The rule avoids the need for all developers on the team to have the contextual understanding of a button's default type.

If you do not wish this rule to apply then using an `eslint-disable` comment or disabling the rule is the most appropriate way forward for you

**Comment by ocket8888** [comment]
On further reflection it may be literally impossible to know if a button is associated with a form, given how complex and dynamic component trees can get. Fair enough.
